### DISCUSSION TITLE
Connectivity to a named instance

### DISCUSSION THREAD

**Discussion by soedro** [trimble-oss/sql-log-shipping-service#109] [Q&A]
Hello,

I have completed a proof of concept with a single db on a regular default instance on the server.

Now I am trying to complete it with restore to a named instance from an AG cluster, and I am facing an issue of connectivity to the Named instance.

I can connect to the Named Instance using SSMS, but when I start the LogShippingService I am getting an error:

...2024-06-17 12:14:56.321 -04:00 [ERR] Error getting databases <55>
Microsoft.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 26 - Error Locating Server/Instance Specified)...
  
This is how the json config file looks like:

"Config": {
    "ContainerUrl": "https://sqlbackups.blob.core.windows.net/ContainerName",
    "SASToken": "encrypted:A==",
    "Destination": "Data Source=Host\\NamedInstance.ourteam.local;Integrated Security=True;Encrypt=True;Trust Server Certificate=True",
    "LogFilePath": "Host\\NamedInstance/{DatabaseName}/LOG/",
    "StandbyFileName": "J:\\Standby\\{DatabaseName}_Standby.BAK",
    "IncludedDatabases": [
      "test"
    ]
  }
}

Any idea on what needs to be changed?

Thanks!

**Comment by DavidWiseman** [comment]
The host should be entered as:
`Data Source=Host.ourteam.local\NamedInstance`
instead of:
`Data Source=Host\NamedInstance.ourteam.local`

If it's a named instance you might need to specify the port number if the browser service isn't running. e.g.
`Data Source=Host.ourteam.local\NamedInstance,12345`

If you are editing the json file directly, the backslash character needs to be replaced with a double backslash as it's a control character in json.  e.g.
```
Data Source=Host.ourteam.local\\NamedInstance

J:\\Standby\\{DatabaseName}_Standby.BAK
etc.
```

From the command line you don't need to double up on backslash as it will encode it correctly for json.
